packages/contracts-bedrock/src/libraries/TransientContext.sol

```diff
@@ -61,10 +61,56 @@ library TransientContext {
 /// @notice Reentrancy-aware modifier for transient storage, which increments and
 ///         decrements the call depth when entering and exiting a function.
 contract TransientReentrancyAware {
+    /// @notice Thrown when a non-written transient storage slot is attempted to be read from.
+    error NotEntered();
+
+    /// @notice Thrown when a reentrant call is detected.
+    error ReentrantCall();
+
+    /// @notice Storage slot for `entered` value.
+    ///         Equal to bytes32(uint256(keccak256("l2tol2crossdomainmessenger.entered")) - 1)
+    bytes32 internal constant ENTERED_SLOT = 0xf53fc38c5e461bdcbbeb47887fecf014abd399293109cd50f65e5f9078cfd025;
```

Comment:
we will want to update the key here to be more generic so this pattern can be used in any contract

Reply:
this good `bytes32(uint256(keccak256("transientreentrancyaware.entered")) - 1)`, or too long?

Reply:
this is good! it just needs to be unique so there is no collision with something that uses this contract